apps/storefront/src/utils/masquerade.ts

```diff
@@ -3,20 +3,24 @@ import {
   superAdminBeginMasquerade,
   superAdminEndMasquerade,
 } from '@/shared/service/b2b';
-import { clearMasqueradeCompany, MasqueradeCompany, setMasqueradeCompany, store } from '@/store';
+import { AppStore, clearMasqueradeCompany, MasqueradeCompany, setMasqueradeCompany } from '@/store';
 
 interface StartMasqueradeParams {
   companyId: number;
   customerId: string | number;
 }
 
-export const startMasquerade = async ({ companyId, customerId }: StartMasqueradeParams) => {
+export const startMasquerade = async (
+  { companyId, customerId }: StartMasqueradeParams,
+  store: AppStore,
```

Comment:
why the argument over turning this into a hook and using `useAppStore` internally?